### PR TITLE
Created SORT function

### PR DESCRIPTION
**Pull request by Kyle1297**
Created SORT excel function.

Linked issue is here: https://github.com/formulajs/formulajs/issues/117

Excel reference: https://support.microsoft.com/en-us/office/sort-function-22f63bd0-ccc8-492f-953d-c20e8e44b86c

Sample Excel sheet of some test cases: 
[sort.xlsx](https://github.com/formulajs/formulajs/files/10149212/sort.xlsx)

### CODE REVIEW

File: src/lookup-reference.js [hunk 1]
```diff
@@ -373,3 +373,145 @@ export function VLOOKUP(lookup_value, table_array, col_index_num, range_lookup)
...
+
+      if (a[sort_index - 1].toString() > b[sort_index - 1].toString()) {
+        return 1 * sort_order
+      }
+
+      return 0
+    })
+  }
+
+  // replace empty strings with zeros
```
[the remark this answers is not in the document]
Comment (Kyle1297) [reply src/lookup-reference.js:507]: I was surprised when I saw this behavior, but for some reason, with the SORT function, empty values are returned with 0

File: src/utils/common.js [hunk 2]
```diff
@@ -365,3 +365,41 @@ export function transpose(matrix) {
 
   return matrix[0].map((col, i) => matrix.map((row) => row[i]))
 }
+
+// E.g. addEmptyValuesToArray([[1]], 2, 2) => [[1, ""], ["", ""]]
+export function addEmptyValuesToArray(array, requiredLength, requiredHeight) {
```
[the remark this answers is not in the document]
Comment (Kyle1297) [reply src/utils/common.js:370]: Seems a little better, but I'm not sure its worth the extra effort

File: src/utils/common.js [hunk 3]
```diff
@@ -365,3 +365,41 @@ export function transpose(matrix) {
...
+  if (!Array.isArray(array) || !array.length) return array;
+  for (let i = 0; i < array.length; i++) {
+    if (!(array[i] instanceof Array)) return array
+  }
+
+  // add empty values to columns
+  for (let i = 0; i < array.length; i++) {
+    if (array[i].length < requiredLength) {
+      for (let j = array[i].length; j < requiredLength; j++) {
+        array[i].push('')
```
[the remark this answers is not in the document]
Comment (Kyle1297) [reply src/utils/common.js:389]: 1. It prevents weird undefined errors if function runs through the arrays still, e.g. with the .toString in the SORT function
2. It doesn't really change anything, does it?

File: src/lookup-reference.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicolashefti) [reply src/lookup-reference.js:502]: I checked. No problem if it is Excel behavior. Can you add a test case to document this kind of edge case ?

File: src/lookup-reference.js [hunk 4]
```diff
@@ -373,3 +373,140 @@ export function VLOOKUP(lookup_value, table_array, col_index_num, range_lookup)
...
+
+    if (array[i].length !== array[0].length) {
+      return error.na
+    }
+  }
+
+  const arrayWidth = array[0].length
+  const arrayHeight = array.length
+
+   if (by_col == null) {
```
Comment (nicolashefti) [inline src/lookup-reference.js:418]: To combine with line ?

`utils.parseBool(by_col || false)`

File: src/utils/common.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolashefti) [reply src/utils/common.js:370]: 1. Renaming/refactoring is made by your IDE
2. The name would be much more explicit to a new developer. It tells cleary what's the function doing
3. Splitting the function in two takes 5 minutes because you wrote already all the test cases

File: src/utils/common.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicolashefti) [reply src/utils/common.js:389]: 1. SORT can be called with _nullish_ (undefined, ...) value. This a common test case in Formulajs.
2. Discussed in the naming comment.

File: src/utils/common.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolashefti) [reply src/utils/common.js:370]: I re-checked, and you are not using this function to build a matrix from an Array anymore. In SORT you are just building a matrix of value (error.value).

In this case just rename the function `matrixFromValue(value, length, height)` and that's it.

File: src/lookup-reference.js [hunk 5]
```diff
@@ -373,3 +373,140 @@ export function VLOOKUP(lookup_value, table_array, col_index_num, range_lookup)
...
+      const column = []
+      for (let j = 0; j < arrayHeight; j++) {
+        column.push(array[j][i])
+      }
+      columns.push(column)
+    }
+
+    const sortedColumns = columns.sort((a, b) => {
+      // NOTE: Excel sorts all values as strings, e.g. 1 => "1"
+      if (a[sort_index - 1].toString() < b[sort_index - 1].toString()) {
```
Comment (nicolashefti) [inline src/lookup-reference.js:468]: You should check if .toString() exists on your value. Or create a parseStringArray (a bit like the existing parseNumberArray).

**Review by nicolashefti — COMMENTED** [review]
Good work. Nicely tested. There are a few details that need to be adressed (at least undefined, null, in the matrix to be sorted).

The helper can be simplified and renamed.
